pkg/utils/tekton/controller.go

```
@@ -841,3 +842,40 @@ func (s *SuiteController) GetTask(name, namespace string) (*v1beta1.Task, error)
 	}
 	return &task, nil
 }
+
+func (s *SuiteController) StorePipelineRuns(componentPipelineRun *v1beta1.PipelineRun, testLogsDir, testNamespace string) error {
+	pipelineRunsDir := fmt.Sprintf("%s/pipeline-runs", testLogsDir)
+	if err := os.MkdirAll(pipelineRunsDir, os.ModePerm); err != nil {
+		return err
+	}
+
+	filepath := fmt.Sprintf("%s/%s-pr-%s.log", pipelineRunsDir, testNamespace, componentPipelineRun.Name)
+	pipelineLogs, err := s.GetPipelineRunLogs(componentPipelineRun.Name, testNamespace)
+	if err != nil {
+		g.GinkgoWriter.Printf("got error fetching PR logs: %v\n", err.Error())
+	}
+
+	if err := os.WriteFile(filepath, []byte(pipelineLogs), 0644); err != nil {
+		g.GinkgoWriter.Printf("cannot write to %s: %+v\n", filepath, err)
```

I think we should return an error here. WDYT?